scripts/o3de/o3de/repo_properties.py

```diff
@@ -263,6 +264,35 @@ def _edit_objects(object_typename:str,
 
     repo_json[f'{object_typename}s_data'] = repo_objects_data
 
+def _auto_update_json(object_type: str or list,
+                      repo_path: pathlib.Path,
+                      repo_json: dict):
+    
+    repo_directory = os.path.dirname(repo_path)
+    expected_files = {"project.json": [],
+                      "gem.json": [],
+                      "template.json": []}
+    for directory, sub_directory, files, in os.walk(repo_directory):
+        found_object_json = False
+        for filename in files:
+            if filename in expected_files.keys():
+                found_object_json = True
+                file_path = pathlib.Path(directory)
+                expected_files[filename].append(file_path)
+        if found_object_json:
+            sub_directory.clear()
```

Comment:
does this prevent the logic from looking into other sub directories once we've found one of the .json files?

Reply:
The Flag statement: 'found_object_json = True' would trigger this sub_directory.clear() so it stops looking at the rest of the directories when the object.json file is found.